app/controllers/admin/suppliers_controller.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+class Admin::SuppliersController < Admin::BaseController
+  skip_after_action :verify_authorized, only: :index
+  skip_after_action :verify_policy_scoped, except: :index
+
+  def index
+    lead_providers = policy_scope(LeadProvider)
+    delivery_partners = policy_scope(DeliveryPartner)
+    sorted_suppliers = (lead_providers + delivery_partners).sort_by(&:name)
+    @suppliers = Kaminari.paginate_array(sorted_suppliers).page(params[:page]).per(20)
+    @page = @suppliers.current_page
+    @total_pages = @suppliers.total_pages
+  end
+
+  def new
+    if params[:continue]
+      @new_supplier_form = NewSupplierForm.new(session[:new_supplier_form])
+    else
+      session.delete(:new_supplier_form)
+      session.delete(:lead_provider_form)
+      @new_supplier_form = NewSupplierForm.new
+    end
+    skip_authorization
+  end
+
+  def receive_new
+    skip_authorization
+    @new_supplier_form = NewSupplierForm.new(params.require(:new_supplier_form).permit(:name))
+
+    render :new and return unless @new_supplier_form.valid?(:name)
+
+    session[:new_supplier_form] = (session[:new_supplier_form] || {}).merge({ name: @new_supplier_form.name })
+    redirect_to admin_new_supplier_type_path
+  end
+
+  def new_supplier_type
+    @new_supplier_form = NewSupplierForm.new(session[:new_supplier_form])
+    skip_authorization
+  end
+
+  def receive_new_supplier_type
+    @new_supplier_form = NewSupplierForm.new(session[:new_supplier_form])
+    @new_supplier_form.type = params.dig(:new_supplier_form, :type)
```

Comment:
What would you prefer?
```
@new_supplier_form = NewSupplierForm.new(session[:new_supplier_form].merge(params[:new_supplier_form]))
```
?

Reply:
```
session[:new_supplier_form].merge!(params[:new_supplier_form])
@new_supplier_form = NewSupplierForm.new(session[:new_supplier_form])
```
Would be nice if it works

Reply:
I don't want it in session until it's been validated

Reply:
Hmm, alright. Then 
```
new_form_params = session[:new_supplier_form].merge(params[:new_supplier_form])
@new_supplier_form = NewSupplierForm.new(new_form_params)
unless @new_supplier_form.valid? (...)

session[:new_supplier_form] = new_form_params
```

I think we should use the default behaviour of matching form accessors to param names as much as possible - hopefully it will mean we don't have to do much param digging ourselves.